src/integrationtests/java/com/aws/greengrass/integrationtests/metrics/MetricsEmitterTest.java

```diff
@@ -0,0 +1,207 @@
+        domainEvents.emit(new AuthorizeClientDeviceActionEvent(AuthorizeClientDeviceActionEvent
+                .AuthorizationStatus.SUCCESS));
+        domainEvents.emit(new VerifyClientDeviceIdentityEvent(VerifyClientDeviceIdentityEvent
+                .VerificationStatus.SUCCESS));
+        domainEvents.emit(new SessionCreationEvent(SessionCreationEvent.SessionCreationStatus.SUCCESS));
+        domainEvents.emit(new ServiceErrorEvent());
+
+        //WHEN
+        startNucleusWithConfig("metricsConfig.yaml");
+        int periodicAggregateIntervalSec = kernel.getContext().get(MetricsConfiguration.class).getAggregatePeriod();
+        assertEquals(AGGREGATE_INTERVAL, periodicAggregateIntervalSec);
+        assertNotNull(kernel.getContext().get(MetricsEmitter.class).getFuture(),
+                "periodic publish future is not scheduled");
+        long delay = kernel.getContext().get(MetricsEmitter.class).getFuture().getDelay(TimeUnit.SECONDS);
+        assertEquals(delay, periodicAggregateIntervalSec);
+
+        //move clock one day ahead to trigger publish
+        Instant aDayLater = Instant.now().plusSeconds(DEFAULT_PUBLISH_INTERVAL);
+        mockInstant(aDayLater.toEpochMilli());
+
+        //check for telemetry logs in ~root/telemetry
+        assertEquals(kernel.getNucleusPaths().rootPath().resolve("telemetry"),
+                TelemetryConfig.getTelemetryDirectory());
+
+        //THEN
+        boolean telemetryMessageVerified = false;
+        String telemetryPublishTopic = DEFAULT_TELEMETRY_METRICS_PUBLISH_TOPIC
+                .replace("{thingName}", MOCK_THING_NAME);
+        CountDownLatch metricsPublished = new CountDownLatch(1);
+        List<PublishRequest> requests = new ArrayList<>();
+        when(mqttClient.publish(any(PublishRequest.class))).thenAnswer(i -> {
+            Object argument = i.getArgument(0);
+            PublishRequest publishRequest = (PublishRequest) argument;
+            if (telemetryPublishTopic.equals(publishRequest.getTopic())) {
+                requests.add(publishRequest);
+                metricsPublished.countDown();
+            }
+            return CompletableFuture.completedFuture(0);
+        });
+        assertTrue(metricsPublished.await(30, TimeUnit.SECONDS), "Metrics not published");
+
+        for (PublishRequest request : requests) {
+            if (!telemetryPublishTopic.equals(request.getTopic())) {
+                continue;
+            }
+            try {
+                MetricsPayload mp = new ObjectMapper().readValue(request.getPayload(), MetricsPayload.class);
+                assertEquals(QualityOfService.AT_LEAST_ONCE, request.getQos());
+                assertEquals("2022-06-30", mp.getSchema());
+                telemetryMessageVerified = true;
+                break;
+            } catch (IOException e) {
+                fail("The message received at this topic is not of MetricsPayload type.", e);
+            }
```

Comment:
this isn't useful. Just allow the test to throw the exception, you don't need try-catch.